### PR TITLE
Fix workflow testing latest featuretools

### PR DESCRIPTION
**Pull request by jeff-hernandez**

### CODE REVIEW

File: .github/workflows/test_latest_repo_change.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+on:
+  schedule:
+    # 05:00 PM UTC / 01:00 PM EST
+    - cron: "0 17 * * *"
+  workflow_dispatch:
+
+env:
+  RECENT: 90000 # seconds
+  WORKFLOW: unit_tests_with_latest_deps.yml
+  GH_TOKEN: ${{ secrets.REPO_SCOPED_TOKEN }}
```
Comment (rwedge) [inline .github/workflows/test_latest_repo_change.yml:12]: is this token used?

Comment (jeff-hernandez) [reply .github/workflows/test_latest_repo_change.yml:12]: yes, the gh cli uses this token to make authorized requests

Comment (jeff-hernandez) [reply .github/workflows/test_latest_repo_change.yml:12]: `REPO_SCOPED_TOKEN` has the permissions necessary to trigger other workflows as opposed to `GITHUB_TOKEN`

File: docs/source/changelog.rst [hunk 2]
```diff
@@ -7,13 +7,14 @@ Future Release
     * Enhancements
         * Added Release Notes CI Check (:pr:`110`)
     * Fixes
+        * Fixed workflow that tests latest changes to featuretools (:pr:`112`)
```
Comment (rwedge) [inline docs/source/changelog.rst:10]: I'd call this a testing change

Comment (jeff-hernandez) [reply docs/source/changelog.rst]: updated

**Review by rwedge — APPROVED** [review]
LGTM

**Pull request merged**